tests/virt_autotest/setup_dns_service.pm

```diff
@@ -0,0 +1,223 @@
+# SUSE's openQA tests
+#
+# Copyright ?? 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Setup dns service for virtual machines to have dns compatiable name.
+# Maintainer: Wayne Chen <wchen@suse.com>
+
+use base "virt_autotest_base";
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $self = shift;
+
+    my $wait_script          = "30";
+    my $bridgeif             = "br123";
+    my $bridgeip             = "192.168.123.1";
+    my $dns_domain_forward   = "testvirt.net";
+    my $dns_domain_reverse   = "123.168.192";
+    my $dns_resolver_file    = "/etc/resolv.conf";
+    my $dns_resolver_tmpfile = "/etc/resolv.tmp";
+    my $dhcpd_config_file    = "/etc/dhcpd.conf";
+    my $dhcpd_service_name   = "dhcpd";
+    my $dhcpd_lease_file     = "/var/lib/dhcp/db/dhcpd.leases";
+    my $dhcpd_lease_time     = "28800";
+
+    assert_script_run("zypper --non-interactive in lsb-release bridge-utils bind bind-chrootenv bind-utils");
+
+    my $get_bridgeif_count = "brctl show | grep $bridgeif | wc -l";
+    my $bridgeif_count     = script_output($get_bridgeif_count, $wait_script, type_command => 1, proceed_on_failure => 1);
+    if ($bridgeif_count eq '0') {
+        assert_script_run("ip link add name $bridgeif type bridge");
+        assert_script_run("ip addr add $bridgeip/24 dev $bridgeif");
+        assert_script_run("ip link set dev $bridgeif up");
+    }
```

Comment:
Maybe you want to make this script more independent, however actually it still reuses the heritage of standalone script, eg the dhcp server setup. Besides this script's network setup inherits what standalone script does to be consistent before and after calling this script, rather than free dynamic setting. So based on this, I would suggest to call standalone script at the beginning and skips the above setting for br123.